tests/Aql/IndexMergeJoin.cpp

```diff
@@ -0,0 +1,337 @@
+    wo.disableWAL = true;
+
+    std::string buffer;
+    buffer.resize(16);
+
+    prefix = basics::hostToBig(prefix);
+    memcpy(buffer.data(), &prefix, 8);
+
+    rocksdb::WriteBatch wb;
+    for (std::size_t i = 0; i < num; i++) {
+      auto v = basics::hostToBig<uint64_t>(generator());
+      memcpy(buffer.data() + 8, &v, 8);
+
+      auto key = rocksdb::Slice(buffer.data(), 16);
+      wb.Put(key, {});
+
+      if (wb.GetDataSize() > 10000) {
+        auto s = db.getDatabase()->Write(wo, &wb);
+        assert(s.Ok());
+        wb.Clear();
+      }
+    }
+
+    auto s = db.getDatabase()->Write(wo, &wb);
+    assert(s.Ok());
+  }
+};
+
+struct JoinTestBase : ::testing::Test {
+  static std::shared_ptr<RocksDBInstance> db;
+};
+
+std::shared_ptr<RocksDBInstance> JoinTestBase::db;
+
+template<typename Pattern>
+struct MyJoinPerformanceTest : JoinTestBase {
+  static void SetUpTestCase() {
+    db = std::make_shared<RocksDBInstance>("foo-bar");
+
+    // generate some data
+    Pattern{}(*db);
+
+    rocksdb::FlushOptions fo;
+    fo.wait = true;
+    db->getDatabase()->Flush(fo);
+  }
+
+  static std::string buildKey(std::uint64_t prefix, std::uint64_t key) {
+    std::string res;
+    res.resize(16);
+    auto v = basics::hostToBig(prefix);
+    memcpy(res.data(), &v, 8);
+    v = basics::hostToBig(prefix);
+    memcpy(res.data() + 8, &v, 8);
+    return res;
+  }
+
+  static void TearDownTestCase() { db.reset(); }
+
+  static std::unique_ptr<rocksdb::Iterator> iterForPrefix(
+      std::uint64_t prefix) {
+    rocksdb::ReadOptions ro;
+    ro.prefix_same_as_start = true;
+    auto key = buildKey(prefix, 0);
+    auto iter =
+        std::unique_ptr<rocksdb::Iterator>(db->getDatabase()->NewIterator(ro));
+    iter->Seek(key);
+    return iter;
+  }
+};
+
+constexpr std::size_t scale = 10000000;
+
+struct SameRangePattern : PatternBase {
+  void operator()(RocksDBInstance& db) {
+    generateData(db, scale, 1, [x = 0]() mutable { return 2 * ++x; });
+    generateData(db, scale, 2, [x = 0]() mutable { return 2 * ++x; });
+  }
+};
+
+struct CommonRangePattern : PatternBase {
+  void operator()(RocksDBInstance& db) {
+    std::size_t total_size = scale;
+    generateData(db, total_size, 1, [x = 0]() mutable { return ++x; });
+    generateData(db, total_size, 2,
+                 [x = total_size / 2]() mutable { return ++x; });
+  }
+};
+
+struct HalfSize : PatternBase {
+  void operator()(RocksDBInstance& db) {
+    std::size_t total_size = scale;
+    // we can assume that the optimizer would pick the smaller collection
+    generateData(db, total_size / 2, 1, [x = 0]() mutable { return 2 * ++x; });
+    generateData(db, total_size, 2, [x = 0]() mutable { return ++x; });
+  }
+};
+
+struct EvenOddPattern : PatternBase {
+  void operator()(RocksDBInstance& db) {
+    generateData(db, scale, 1, [x = 0]() mutable { return 2 * ++x; });
+    generateData(db, scale, 2, [x = 0]() mutable { return 2 * ++x + 1; });
+  }
+};
+
```

Comment:
```
struct BigGapsPattern : PatternBase {
  void operator()(RocksDBInstance& db) {
    generateData(db, scale, 1, [x = uint64_t{0}]() mutable { return ++x; });
    generateData(db, scale / 100, 2, [x = uint64_t{0}]() mutable { return ++x * 1000; });
  }
};
```

Reply:
To make this fair, the first one has to be the smaller collection. The optimizer will always try to use the smaller collection first.

Reply:
Understand. The idea was to check case when you have to pretty sparse match group.

Btw if this PR compare seek vs next approach, and seek overhead is big, maybe something hybrid will be good? Do Next some n times before Seek?
If "swap" or "equal" happened reset the counter.